kratos/includes/model_part.h

```diff
@@ -800,56 +800,85 @@ class KRATOS_API(KRATOS_CORE) ModelPart : public DataValueContainer, public Flag
     bool HasProperties(IndexType PropertiesId, IndexType ThisIndex = 0) const
     {
         auto pprop_it = GetMesh(ThisIndex).Properties().find(PropertiesId);
-        if(pprop_it != GetMesh(ThisIndex).Properties().end()) { //property does exist
+        if(pprop_it != GetMesh(ThisIndex).Properties().end()) { // Property does exist
             return true;
         }
 
         return false;
     }
 
-    /** Returns the Properties::Pointer  corresponding to it's identifier */
+    /**
+     * @brief Creates a new property in the current mesh
+     * @details If the property is already existing it will crash
+     * @param PropertiesId The Id of the new property
+     * @param ThisIndex The Id of the mesh (0 by default)
+     * @return The new created properties
+     */
+    PropertiesType::Pointer CreateNewProperties(IndexType PropertiesId, IndexType ThisIndex = 0)
+    {
+        auto pprop_it = GetMesh(ThisIndex).Properties().find(PropertiesId);
+        if(pprop_it != GetMesh(ThisIndex).Properties().end()) { // Property does exist
+            KRATOS_ERROR << "Property already existing. Please use pGetProperties() instead" << std::endl;
+        } else {
+            if(IsSubModelPart()) {
+                PropertiesType::Pointer pprop =  mpParentModelPart->CreateNewProperties(PropertiesId, ThisIndex);
+                GetMesh(ThisIndex).AddProperties(pprop);
+                return pprop;
+            } else {
+                PropertiesType::Pointer pnew_property = Kratos::make_shared<PropertiesType>(PropertiesId);
+                GetMesh(ThisIndex).AddProperties(pnew_property);
+                return pnew_property;
+            }
+        }
+
+        return nullptr;
```

Comment:
this is not used

Either this function throws or it returns a valid pointer

Reply:
Yes, but if I don't put it I have warnings for not returning anything (some compilers don't detect it properly)

Reply:
@loumalouomega are u really sure this is needed? i would say you can never get here because of the if else structure